papers/Instruction_Mining_When_Data_Mining_Meets_Large_Language_Model_Finetuning.md

```diff
@@ -0,0 +1,52 @@
+# [Instruction Mining: When Data Mining Meets Large Language Model Finetuning](https://arxiv.org/abs/2307.06290)
+
+ID: 34
+
+## Summary
+
+This paper seeks to find an optimal subset to instruction-tune an LLM with. They do this by considering a bunch of quality and diversity metrics:
+- Input length
+- Output length
+- Reward score from a 1.4B Pythia model trained on open assistant data
+- Perplexity
+- MTLD: Measure of Textual Lexical Diversity [1]
+- KNN-i: Distance to i-th nearest neighbours in SentenceBERT embedding space
+- Unieval-naturalness: Whether a response is like something a person would naturally say [2]
+- Unieval-coherence: Whether a response is a valid continuation of the previous conversation [2]
+- Unieval-understandability: Whether a response is understandable [2]
+and then gather a dataset by training a bunch of Llama-2-7Bs on different data mixes, and recording the QD metrics and eval losses. With this they find a formula to predict eval loss from these metrics and use BlendSearch [3] to find optimal subsets.
+
+Based on their analysis the only relevant metrics are the quality metrics reward score and the Unieval scores, so diversity as measured by MTLD or KNN-i doesn't play a role.
+
+They find that the optimal size is rather small, around 2k samples. They call this double descent, but this has nothing to do with how that term is used normally in the literature.
+
+## Relevance to survey topic (1-5)
+
+Relevance: 5
+
+## Algorithms
+
+- BlendSearch [3]
+
+## Benchmarks
+
+- MT-bench loss
+- OpenLLM benchmarks
+- GPT-as-a-judge (not sure what dataset they used for this)
+
+## Metric Results
+
+There is a significant improvement over random subset selection at a relatively small cost. They show
```

Comment:
Testing is done relatively in-distribution.

Reply:
Added to paper description